### PR TITLE
[terra-button-group] Updated tabIndex on button group for single select example 

### PR DESCRIPTION
**Pull request by KV106606Viswanath**
### Summary
Previously, we are unable to tab (using tab key) between buttons on the button group in single select example. With this fix it has been resolved and we can move focus to the button even it is not selected on single select example. 

**What was changed:**
Updated tabIndex on button group for single select example. 

**Why it was changed:**
User unable to tab between buttons on button group.

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10171

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by saket2403** [comment]
Keyboard navigation for single select buttons should be similar to radio button fields ie: Tab should put focus on the first button and subsequent tabbbing should move the focus out of the group to the next focusable element. Only arrow keys should work inside the group for single select buttons.Removing the entire tabIndex logic means that functioanlity is breaking.

**Comment by KV106606Viswanath** [comment]
> Keyboard navigation for single select buttons should be similar to radio button fields ie: Tab should put focus on the first button and subsequent tabbbing should move the focus out of the group to the next focusable element. Only arrow keys should work inside the group for single select buttons.Removing the entire tabIndex logic means that functioanlity is breaking.

Updated.

**Comment by saket2403** [comment]
Focus should come onto the selected item (when present) initially but as per these changes, it will always come onto the first button.

**Comment by KV106606Viswanath** [comment]
> Focus should come onto the selected item (when present) initially but as per these changes, it will always come onto the first button.

Updated

File: packages/terra-button-group/src/ButtonGroup.jsx [hunk 1]
```diff
@@ -170,7 +170,10 @@ class ButtonGroup extends React.Component {
 
     React.Children.forEach(children, (child, index) => {
       const isSelected = selectedKeys.indexOf(child.key) > -1;
-      const tabAttr = (this.props.onChange && !isMultiSelect) ? { tabIndex: (isSelected) ? '0' : '-1' } : null;
+      // Checking whehter any option is selected or not
+      const isAnyOptionSelected = children && Array.isArray(children) ? children.findIndex((e) => this.props.selectedKeys.includes(e?.key)) : -1;
```
Comment (saket2403) [inline packages/terra-button-group/src/ButtonGroup.jsx:174]:
```suggestion
      const isAnyOptionSelected = children && Array.isArray(children) ? children.findIndex((child) => this.props.selectedKeys.includes(child?.key)) : -1;
```

**Comment by saket2403** [comment]
Please add a wdio test to verify that focus moves to the first button when no buttons are selected initially in a single select button group.

**Comment by KV106606Viswanath** [comment]
> Please add a wdio test to verify that focus moves to the first button when no buttons are selected initially in a single select button group.

Added.

File: packages/terra-button-group/src/ButtonGroup.jsx (hunk 1 above)
Comment (KV106606Viswanath) [reply packages/terra-button-group/src/ButtonGroup.jsx]: Updated

File: packages/terra-button-group/src/ButtonGroup.jsx [hunk 2]
```diff
@@ -170,7 +170,10 @@ class ButtonGroup extends React.Component {
 
     React.Children.forEach(children, (child, index) => {
       const isSelected = selectedKeys.indexOf(child.key) > -1;
-      const tabAttr = (this.props.onChange && !isMultiSelect) ? { tabIndex: (isSelected) ? '0' : '-1' } : null;
+      // Checking whehter any option is selected or not
```
[the remark this answers is not in the document]
Comment (KV106606Viswanath) [reply packages/terra-button-group/src/ButtonGroup.jsx]: Updated

File: packages/terra-button-group/src/ButtonGroup.jsx [hunk 3]
```diff
@@ -170,7 +170,11 @@ class ButtonGroup extends React.Component {
 
     React.Children.forEach(children, (child, index) => {
       const isSelected = selectedKeys.indexOf(child.key) > -1;
-      const tabAttr = (this.props.onChange && !isMultiSelect) ? { tabIndex: (isSelected) ? '0' : '-1' } : null;
+      // Checking whether any option is selected or not
+      // eslint-disable-next-line no-shadow
```
Comment (saket2403) [inline packages/terra-button-group/src/ButtonGroup.jsx:174]: Why is this eslint rule disabled?

**Review by saket2403 — APPROVED** [review]

File: packages/terra-button-group/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Removed
+  * Removed `tabAttr` variable to fix tabbing issue on the button group.
```
Comment (sugan2416) [inline packages/terra-button-group/CHANGELOG.md:6]:
```suggestion
* Fixed
  * Fixed tabbing issue for button-group without initial focus.
```

File: packages/terra-core-docs/CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 
 * Added
+  * Added a test example to validate focus when none of the buttons are selected.
```
Comment (sugan2416) [inline packages/terra-core-docs/CHANGELOG.md:6]:
```suggestion
  * Added a test example for `terra-button-group` to validate focus when none of the buttons are selected.
```

Comment (KV106606Viswanath) [reply packages/terra-core-docs/CHANGELOG.md]: Updated

File: packages/terra-button-group/CHANGELOG.md (hunk 4 above)
Comment (KV106606Viswanath) [reply packages/terra-button-group/CHANGELOG.md]: Updated

**Review by sugan2416 — APPROVED** [review]

**Comment by rahulkumar8cs** [comment]
+1

**Pull request merged**